Use simctl to take screenshots by default

dpgraham opened the pull request.

imurchie commented on test/functional/basic/screenshot-e2e-specs.js:
```diff
@@ -0,0 +1,33 @@
+  });
+
+  it('should take screenshot', async () => {
+    await driver.takeScreenshot().should.eventually.be.a.string;
```
https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/basic/basic-e2e-specs.js#L86-L111

dpgraham replied:
Aha, I'll move the tests over to that file

imurchie commented on lib/commands/screenshot.js:
```diff
@@ -0,0 +1,19 @@
+  try {
+    screenshot = await getScreenshot(this.opts.udid);
+  } catch (e) {
+    return this.proxyCommand('/screenshot', '/GET');
```
Some sort of logging of the error would be good, for debugging's sake.

imurchie replied:
Also, `return await`, or, to make it of the same sort as the previous call, `screenshot = await this.proxyCommand('/screenshot', '/GET');`.

dpgraham commented:
Changes applied

* Added logger that indicates when legacy screenshot is being used
* Removed 'screenshot.js' and moved everything into 'screenshots.js'
* Removed old test and just use the one that is already there
* Doesn't let it fail anymore, just checks that xcode version is >= 8.2

imurchie commented on lib/commands/screenshots.js:
```diff
@@ -1,17 +1,23 @@
-      throw new Error(`Unable to take screenshot. WDA returned '${JSON.stringify(data)}'`);
-    }
-  });
+  if (this.xcodeVersion.major > 8 || (this.xcodeVersion.major === 8 && this.xcodeVersion.minor >= 1)) {
```
This could also just be `if (this.xcodeVersion.versionFloat >= 8.1) {`.

dpgraham commented:
Changes applied

dpgraham commented:
Changes published to 2.25.1
